Dining/views.py

```diff
@@ -51,154 +84,84 @@ def process_date(context, day, month, year):
-        if next_date is not None:
-            self.context['next_link'] = reverse_day('day_view', next_date)
-        else:
-            self.context['next_link'] = None
+        context['next_date'] = self.date + timedelta(days=3 if self.date.weekday() == 4 else 1)
+        context['previous_date'] = self.date - timedelta(days=3 if self.date.weekday() == 0 else 1)
 
-        prev_date = self.get_previous_availlable_date(current_date)
-        if prev_date is not None:
-            self.context['prev_link'] = reverse_day('day_view', prev_date)
-        else:
-            self.context['prev_link'] = None
+        context['dining_lists'] = DiningList.get_lists_on_date(self.date)
+        context['Announcements'] = DiningDayAnnouncements.objects.filter(date=self.date)
 
-        self.context['add_link'] = reverse_day('new_slot', current_date)
-        self.context['dining_lists'] = DiningList.get_lists_on_date(current_date)
-        self.context['Announcements'] = DiningDayAnnouncements.objects.filter(date=current_date)
+        # Check if create slot button must be shown
+        # Todo: optionally hide claim button when time is past closure time
+        # (but I prefer to only check when claiming to reduce code)
+        in_future = self.date >= timezone.now().date()
+        context['can_create_slot'] = DiningList.objects.available_slots(self.date) >= 0 and in_future
 
-        self.context['can_create_slot'] = False
-        if current_date.weekday() > 4:
-            self.context['message'] = "Kitchen can't be used on weekends"
-        else:
-            slot_limit = DiningDayAnnouncements.objects.filter(date=current_date).aggregate(Sum('slots_occupy'))
-            if slot_limit['slots_occupy__sum'] is None:
-                slot_limit = 0
-            else:
-                slot_limit = slot_limit['slots_occupy__sum']
-
-            if len(self.context['dining_lists']) < MAX_SLOT_NUMBER - slot_limit:  # if maximum slots is not exceeded
-                if current_date > timezone.now().date():  # if day is in the future
-                    self.context['can_create_slot'] = True
-                elif (current_date - timezone.now().date()).days == 0:  # if date is today
-                    from .constants import DINING_SLOT_CLAIM_CLOSURE_TIME
-                    if timezone.now().time() < DINING_SLOT_CLAIM_CLOSURE_TIME:  # if it's not past 17:00
-                        self.context['can_create_slot'] = True
-
-        self.context['interactive'] = True
-
-        return render(request, self.template, self.context)
-
-    @staticmethod
-    def get_next_availlable_date(current_date):
-        max_future = 7  # Define the meaximum time one can go into the future
-
-        next_date = current_date + timedelta(days=1)
-        while next_date.weekday() > 4:  # i.e. 5 or 6 which is saturday or sunday
-            next_date = next_date + timedelta(days=1)
+        # Don't know what this is
+        context['interactive'] = True
```

Comment:
Het is een indicatie of de slots klikbaar zijn of slechts visueel. In de dining list is slot weergave niet interactive en dus niet klikbaar.